[POA-41] Update analytics functions to support amplitude functions

mudit-postman opened the pull request:
* Replace Segment related keys and mentioned with Amplitude Keys and configs
* Updated few event names to make them consistent in Amplitude

mudit-postman commented on telemetry/telemetry.go:
```diff
@@ -114,7 +114,7 @@ func getDistinctID() string {
 	// Failing that, try to use the user name and host name?
 
-	id := os.Getenv("AKITA_SEGMENT_DISTINCT_ID")
+	id := os.Getenv("POSTMAN_ANALYTICS_DISTINCT_ID")
```
Do we need to do any changes for this in documentations?
cc @mgritter @liujed

mgritter replied:
No, this is not a documented feature.  The ability to disable analytics is more important.

mgritter commented on telemetry/telemetry.go:
```diff
@@ -24,7 +24,7 @@ var (
 	analyticsEnabled bool
 
 	// Client key; set at link-time with -X flag
```
Needs a corresponding change in superstar/cli-postman and superstar/cli-akita

mgritter commented on telemetry/telemetry.go:
```diff
@@ -83,23 +83,23 @@ func Init(isLoggingEnabled bool) {
-			// No output from the Segment library
+			// No output from the Amplitude library
 			IsLoggingEnabled: false,
-			// IsMixpanelEnabled: false,  -- irrelevant for us, leaving at default value
-			BatchSize: 1, // disable batching
```
I don't recall precisely why I disabled batching, but I think it had to do with catching errors during command-line execution, or maybe other abnormal exits.  I think that should still be part of our use of Amplitude, if batching can be disabled?

mudit-postman replied:
We can disable batching we can use `FlushQueueSize` config for this.

mgritter commented on telemetry/telemetry.go:
```diff
@@ -255,10 +255,10 @@ func APIError(method string, path string, e error) {
 func Failure(message string) {
 	analyticsClient.Track(distinctID(),
-		message,
+		fmt.Sprintf("Unknown Error: %s", message),
```
I am unsure about the need to add the prefixes as you have done here.  I don't think we really felt their absence using Mixpanel, and the "type" field could be used to filter.  Is this an Amplitude or Postman best practice that we should understand?

We should not use "Unknown Error" here, though, because the failures can be known errors that simply don't come with a Go "error".

mudit-postman replied:
I have just added these prefixes to follow a convention with the event names, also it can help us in the easier resolution of new events that might be added in the Amplitude.
Amplitude doesn't automatically start tracking a new event, we need to tell Amplitude to start tracking the event. Like in the attached image, events that are not being tracked are marked as Unexpected we need to add them to the list to make them live. 
https://app.amplitude.com/data/postman/Postman/events/POA-41-migrate-to-amplitude/latest?view=All
<img width="1219" alt="image" src="https://github.com/akitasoftware/akita-cli/assets/116863733/35dabbf1-5524-4b6c-b4fe-2c72b9ac0a34">

mgritter commented on telemetry/telemetry.go:
```diff
@@ -268,7 +268,7 @@ func Failure(message string) {
 func Success(message string) {
 	analyticsClient.Track(distinctID(),
-		message,
+		fmt.Sprintf("Success in %s", message),
```
For example, in this case we have only two instances:

"Success in rotate learn session"
"Success in Add to ECS"

which is inconsistent capitalization :) and I don't think the success adds much?

mudit-postman commented:
Fixed `backend_collector_test.go` test case.
We have updated the akita-libs version here which has changes related to obfuscation [PR](https://github.com/akitasoftware/akita-libs/pull/196), due to which test case was failing
